client/state/routing/middleware.js

```diff
@@ -9,11 +9,18 @@ import page from 'page';
  */
 import localforage from 'lib/localforage';
 import { isOutsideCalypso } from 'lib/url';
-import { ROUTE_SET } from 'state/action-types';
+import { ROUTE_SET, ROUTE_TO } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
 const LAST_PATH = 'last_path';
 
+export const routingMiddleware = () => ( next ) => ( action ) => {
+	if ( action.type === ROUTE_TO ) {
+		page( action.path );
+	}
```

Comment:
Separating out the PRs is good I think.

Adding this routing middleware is one of those "small tasks" that's been on my list for some time now but I haven't gotten around to it.

As far as sharing code goes, I'm fine letting some of this slide. Ultimately I think we'll have the best option if we have a decoder/encoder pair for the section.

```js
const fromPath = path => {
	const match = pathMatch( '/comments/[:filterName]/:siteId' );
	if ( ! match ) {
		return {
			filterName: 'all',
			siteId: null,
		}
	}

	const { filterName: rawFilter, siteId: rawSite } = match;
	const filterName = includes( filterNames, rawFilter ) ? rawFilter : 'all';
	const siteId = parseInt( rawSite, 10 );

	return Object.assign(
		{ filterName},
		siteId && { siteId },
	);
}

cont toPath = ( { filterName, siteId } ) => {
	const filterPart = filterName ? `/${ filterName }` : '';
	const sitePart = siteId ? `/${ siteId }` : '';

	return `/comments${ filterPart }${ sitePart }`;
}
```

at least, this is what I think is great because of how it gives us control over the URL and how we have a clear and controlled way of translating path structure into app state (for example, when the path is messed up).